DateSense/DSoptions.py

```diff
@@ -300,7 +300,7 @@ def initialize(self, dates):
             the formatting should be detected.
         '''
         # If it's just one string, turn it into a collection like the methods expect
-        if isinstance(dates, str):
+        if isinstance(dates, ("".__class__, u"".__class__)):
```

Comment:
This is a clever hack that someone on Stackoverflow came up with, but I think explicit is better than implicit. Code golf is not necessary here.

Reply:
There is no more explicit way to perform this check without adding a new external dependency. The alternative to this and to adding another dependency is to catch a `NameError` when `basestring` is not defined in Python 3 and check for `str` instead. The `NameError` solution would be unclear to anyone who was not familiar with why checking if things are strings differs in Python 2 and 3. This solution is very clear - Is this object an instance of the same class that string literals are instances of?

Reply:
@pineapplemachine correct me if I'm wrong, but I think the main reason for @dwyerk's change request is ensuring full Python 2 support.  I get that it works in Python 3 and most of the time in Python 2, but isn't `isinstance(dates, str)` insufficient for checking if something is a string in Python 2?  With the current code, someone using Python 2 could pass in `u'2018-01-01T00:00:00Z'` as the `dates` variable and the `isinstance(dates, str)` check would return `False` when we actually want it to return `True`.  

An alternative approach might be to check if `basestring` is defined, and if not, set it equal to `str`, then keep this line unchanged.  I don't think that's better than @dwyerk's suggestion, though. 
 Thoughts?

Reply:
> With the current code, someone using Python 2 could pass in `u'2018-01-01T00:00:00Z'` as the `dates` variable and the `isinstance(dates, str)` check would return `False` when we actually want it to return `True`.

@sfairgrieve, that is not correct. Here is an example from my Python 2.7.9 shell.

``` python
>>> isinstance("binary string", ("".__class__, u"".__class__))
True
>>> isinstance(u"unicode string", ("".__class__, u"".__class__))
True
```

Here is an example of DateSense usage with Python 2.7:

``` python
print(DateSense.detect_format(u"2018-04-26").get_format_string()) # correctly prints u"%Y-%m-%d"
```

Reply:
@pineapplemachine You're right.  I misread the comment/history here.  Disregard.

Reply:
I also after re-reading see that you may have meant something different from saying that the PR change would fail for unicode inputs in Python 2.7? Confusion all around, I guess.